src/data/viteComboFetcher.ts

```diff
@@ -0,0 +1,134 @@
+
+  // issues define the ID of the Combo and an issue message
+  const issues: [number, string][] = []
+
+  allCombos.forEach( combo => {
+    if(combo.tricks) {
+      combo.tricks.forEach( (f,i) => {
+        if(f[1] !== "official") {
+          issues.push([combo.id, `All tricks must limit themselves to official ones (not the case for ${i+1}. Entry)`])
+        }
+        else if(!idToTrickLookup[f[0]]) {
+          issues.push([combo.id, `${i+1}. Entry in tricks references an ID that does not exist (id: ${f[0]}).`])
+        }
+
+      })
+    }
+  })
+
+  return issues;
+}
+
+
+
+
+
+export default async function viteGetAllCombos(tricks: z.infer<typeof DbTricksTableZod>[]) {
+  // @ts-expect-error Because there are effectively two TS Projects (Vite Plugin Context 
+  //and Vue Webapp Context), TS gets a bit confused and doesn't think import.meta.url is allowed here
+  const pattern = join(fileURLToPath(import.meta.url), "..", "combos", "*.yaml");
+  const allFilePaths = await globby(pattern)
+
+  const parsedYamlFiles = await Promise.allSettled(allFilePaths.map( comboFilePath => fetchYamlFile(comboFilePath)))
+  const erroredFiles = parsedYamlFiles.map( (fileResult,i) => fileResult.status === "rejected" ? ({error: fileResult.reason, path: allFilePaths[i]}) : (undefined as never) ).filter(Boolean);
+  const goodFiles = parsedYamlFiles.map( fileResult => fileResult.status === "fulfilled" ? fileResult.value : (undefined as never)).filter(Boolean)
+
+  if(erroredFiles.length) {
+
+    const errorMeta = erroredFiles.map(file => {
+      if(!(file.error instanceof ZodError)) {
+        return file.error+""
+      }
+      return file.path+"\n"+file.error.errors.map( error => `${error.path}: ${error.message}`).join("\n")
+    })
+
+    throw {
+      plugin: "vite-plugin-highline-freestyle-data",
+      message: "At least one Combo file could not be parsed.\n\n"+errorMeta,
+    }
+  }
+
+  const duplicateIds = findDuplicateKeys(goodFiles)
+  if(duplicateIds.length > 0) {
+    throw {
+      plugin: "vite-plugin-highline-freestyle-data",
+      message: "Some Combos contain the same ID.\n\n"+JSON.stringify(duplicateIds)
+    }
+  }
+
+  const emptyCombos = goodFiles.filter(combo => !combo.tricks || combo.tricks.length === 0).map( combo => ([combo.id, "Combo needs at least 1 trick"] as const))
+  if(emptyCombos.length > 0) {
+    throw {
+      plugin: "vite-plugin-highline-freestyle-data",
+      message: "Empty Combos detected.\n\n"+emptyCombos.map( ([comboId, errorMessage]) => `Failed Combo: ${comboId}. Issue: ${errorMessage}`).join("\n")
+    }
+  }
+
+  const undefinedReferences = findUndefinedTrickReferences(goodFiles, tricks)
+  if(undefinedReferences.length > 0) {
+    throw {
+      plugin: "vite-plugin-highline-freestyle-data",
+      message: "Reference Resolution failed.\n\n"+undefinedReferences.map( ([comboId, errorMessage]) => `Failed Combo: ${comboId}. Issue: ${errorMessage}`).join("\n")
```

Comment:
Maybe 'Undefined reference found.' (or something similar) instead of 'reference resolution failed.' ?

Reply:
Changed to `Undefined trick reference found`